src/auth_service/auth_adapter/api/main.py

```diff
@@ -311,26 +325,38 @@ async def create_new_totp_token(
     status_code=status.HTTP_204_NO_CONTENT,
 )
 async def rpc_verify_totp(  # noqa: PLR0913
-    verification_info: VerifyTOTP,
     session_store: SessionStoreDependency,
     session: SessionDependency,
     totp_handler: TOTPHandlerDependency,
     user_registry: Annotated[UserRegistryPort, Depends(get_user_registry)],
     token_dao: Annotated[UserTokenDao, Depends(get_user_token_dao)],
+    x_authorization: Annotated[
+        str | None, Header(title="the TOTP code to verify")
+    ] = None,
 ) -> Response:
-    """Verify a TOTP token."""
+    """Verify a TOTP token.
+
+    The TOTP code is expected to be passed like a bearer token
+    in the "X-Authorization" header.
+
+    Note: Since this endpoint is used via the ExtAuth protocol,
+    the request body cannot be used to pass the TOTP code.
+    """
     if not session:
         raise HTTPException(
             status_code=status.HTTP_401_UNAUTHORIZED, detail="Not logged in"
         )
-    if not session.user_id or session.user_id != verification_info.user_id:
+
+    totp = get_bearer_token(x_authorization)
+    if totp:
+        totp = totp.removeprefix("TOTP:")
```

Comment:
If the X-Authorization header does not have the "TOTP:" prefix, it will not be removed, but the totp variable will still hold the original value instead of totp, e.g., JWT might be included in the header instead of TOTP.
An early check like this may be useful: `if not totp or not totp.startswith("TOTP:")`

Reply:
That's intended, I wanted to make this "TOTP:" prefix optional. It serves only to indicate that this is not your normal JWT.